clients/roscpp/src/libros/timer.cpp

```diff
@@ -95,7 +95,7 @@ Timer::Timer(const TimerOptions& ops)
   impl_->callback_ = ops.callback;
   impl_->callback_queue_ = ops.callback_queue;
   impl_->tracked_object_ = ops.tracked_object;
-  impl_->has_tracked_object_ = ops.tracked_object;
+  impl_->has_tracked_object_ = (bool)ops.tracked_object;
```

Comment:
I think it might be more idiomatic of C++ and safer to use `static_cast<bool>(...)`.